src/firefly/js/visualize/saga/UrlLinkWatcher.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+
+import {TABLE_LOADED} from '../../tables/TablesCntlr.js';
+import {findTableAccessURLColumn} from '../../util/VOAnalyzer.js';
+
+
+/** type {TableWatcherDef}
+ *  TODO: for testTable, more logic could be added if the API of setting table meta for any link-like column is supported
+ * */
+export const urlLinkWatcherDef = {
+    id : 'URLLinkWatcher',
+    watcher : watchURLLinkColumns,
+    testTable : (table) => findTableAccessURLColumn(table),
+    actions: [TABLE_LOADED]
+};
+
+
+
+/**
+ * Action watcher callback: watch the tables with access_url column and update table column attributes for
+ * table rendering which displays access_url column as a link.
+ * @param tbl_id
+ * @param action
+ * @param cancelSelf
+ * @param params
+ * @return {*}
+ */
+export function watchURLLinkColumns(tbl_id, action, cancelSelf, params) {
+    if (!action) {
+        return params;
+    }
+    const {payload}= action;
+    if (payload.tbl_id && payload.tbl_id!==tbl_id) return params;
+
+    switch (action.type) {
+        case TABLE_LOADED:
+            handleLinkColumnUpdate(tbl_id);
+            break;
+    }
+    return params;
+}
+
+function handleLinkColumnUpdate(tbl_id) {
+    const linkCol = findTableAccessURLColumn(tbl_id);
+
+    if (linkCol && (!linkCol.type || linkCol.type === 'char')) {
+        linkCol.type = 'location';
```

Comment:
This is store data which should be treated as immutable. Is that not a dispatch that you can change it with?

Reply:
Thank you for the catching!